### PR TITLE
add base62.Encode and base62.Decode function

### PR DESCRIPTION
**Pull request by yanyanqing**

### CODE REVIEW

File: pkg/base62/base62.go [hunk 1]
```diff
@@ -0,0 +1,60 @@
+package base62
+
+import "math"
+
+const ary = 62
```
Comment (IANTHEREAL) [inline pkg/base62/base62.go]: rename `ary` to `base`

File: pkg/base62/base62.go [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+// Encode encodes m to a string
+func Encode(m int64, length int64) string {
+	arr := make([]int64, length)
```
Comment (IANTHEREAL) [inline pkg/base62/base62.go]: rename `arr` to `bits`

File: pkg/base62/base62.go [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+	i := 0
```
Comment (IANTHEREAL) [inline pkg/base62/base62.go]: `i := length-1`

File: pkg/base62/base62.go (hunk 2 above)
Comment (yanyanqing) [reply pkg/base62/base62.go]: ok

File: pkg/base62/base62.go [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+	}
+
+	return encodeCore(arr, length)
+}
+
+func encodeCore(arr []int64, length int64) string {
+	var result string
+	var value int64
+	for _, v := range arr {
+		if v < 0 {
```
Comment (IANTHEREAL) [inline pkg/base62/base62.go]: it wouldn't happen.  seem useless

File: pkg/base62/base62.go (hunk 3 above)
Comment (yanyanqing) [reply pkg/base62/base62.go]: why ？

Comment (IANTHEREAL) [reply pkg/base62/base62.go]: Reverse order

**Review by yanyanqing — COMMENTED** [review]
@WangXiangUSTC 
@LinuxGit  Gangster review for me

**Comment by LinuxGit** [comment]
LGTM

**Comment by IANTHEREAL** [comment]
LGTM

**Pull request merged**